card/mod/carrierwave/lib/carrier_wave/file_card_uploader.rb

```diff
@@ -159,6 +159,10 @@ class FileCardUploader < Uploader::Base
     ].freeze
     delegate :store_dir, :retrieve_dir, :file_dir, :mod, :bucket, to: :model
 
+    def valid?
+      extension.present?
```

Comment:
what about files without extension?

Reply:
Currently we can't handle those.  We generate a url that ends in a period (eg `files/~12/345.`), and then decko misinterprets the url as requesting the file in html format.  This translates into fairly expensive requests.

I'll add a note that says it should eventually be valid for files not to have an extension.  But right now I think the only way you'd actually be able to access such a file is (maybe?) with the `.file` extension, which is weird decko functionality almost nobody knows about.

Reply:
I ended up not using this check for now, because it was killing a bunch of wikirate specs where we didn't have valid images (logo, Samsung+image, missing, etc).  The check is still here but not used in source view anymore, so for now it does nothing.

Reply:
hmm correction: I guess it's still used on File cards but not Images